nginx-controller/controller/controller.go

```diff
@@ -1222,6 +1223,30 @@ func (lbc *LoadBalancerController) getMinionsForMaster(master *nginx.IngressEx)
 			glog.Errorf("Ingress Resource %v/%v with the 'nginx.org/mergible-ingress-type' annotation set to 'minion' must contain a Path", ings.Items[i].Namespace, ings.Items[i].Name)
 			continue
 		}
+
+		for key, val := range ings.Items[i].Annotations {
+			if key == "nginx.org/mergible-ingress-type" || !strings.Contains(key, "nginx.org/") {
+				continue
+			}
+			if !strings.Contains(val, "serviceName=") {
```

Comment:
I was thinking minion annotations should only contain annotations where serviceName is applied.

We can make "serviceName=" a global since it is used in multiple places.

Reply:
I think we should handle most of the annotations that could be applied per a location block in the NGINX config or  per an upstream (service). Such a handling enables new use cases.

For example, consider an Ingress resource with multiple paths and multiple services. If we want to choose a load balancing method, other than the default round robin, we need to use `nginx.org/lb-method` annotation. However, this annotation applies to all services in the Ingress resource. What if we want to change the lb method only for one service. To overcome this limitation, we could potentially change the `nginx.org/lb-method` to support specifying a service, similar to what we did for [session persistence](https://github.com/nginxinc/kubernetes-ingress/tree/master/examples/session-persistence). However, with mergible Ingresses, we can split the Ingress resource into multiple ones and in each minion Ingress resource have only one path/service and configure the lb method using the `nginx.org/lb-method` annotation for each service.

Another example is `nginx.org/location-snippets` annotation.

Reply:
That works for me. So for example:

Minion_1 contains `/tea` and `/coffee`. If we want to change the load-balancing for both then we apply `nginx.org/lb-method` which configures both. Now if we wanted it only for `/coffee` then we split it up into 2 minions and apply it only to the 1 with `/coffee`.

I can remove the `serviceName` check and then apply all the master's annotations to the minions and then from each minion apply only their specific annotations overwriting the masters. All within the functions which generates the Nginx Configuration.